app/src/server/api/routers/datasets.router.ts

```diff
@@ -53,55 +55,67 @@ export const datasetsRouter = createTRPCRouter({
   getTrainingCosts: protectedProcedure
     .input(
       z.object({
-        id: z.string(),
+        datasetId: z.string(),
+        baseModel: baseModel,
         filters: filtersSchema,
-        selectedPruningRuleIds: z.array(z.string()),
+        pruningRuleIds: z.array(z.string()),
       }),
     )
     .query(async ({ input, ctx }) => {
-      const { id, filters, selectedPruningRuleIds } = input;
+      const { datasetId, filters, baseModel, pruningRuleIds } = input;
 
       const { projectId } = await prisma.dataset.findUniqueOrThrow({
-        where: { id },
+        where: { id: datasetId },
       });
 
       await requireCanViewProject(projectId, ctx);
 
-      const trainingEntryStats = await constructDatasetEntryFiltersQuery({ filters, datasetId: id })
+      const datasetEntryStats = await constructDatasetEntryFiltersQuery({ filters, datasetId })
         .where("de.split", "=", "TRAIN")
         .where("de.output", "is not", null)
-        .select((eb) => [
-          sql<number>`count(*)::int`.as("numEntries"),
+        .select([
+          sql<number>`count(distinct de.id)::int`.as("numEntries"),
           sql<number>`sum(de."inputTokens")::int`.as("totalInputTokens"),
           sql<number>`sum(de."outputTokens")::int`.as("totalOutputTokens"),
-          eb
-            .selectFrom("PruningRuleMatch as prm")
-            .whereRef("prm.datasetEntryId", "=", "de.id")
-            .where("prm.pruningRuleId", "in", selectedPruningRuleIds)
-            .leftJoin("PruningRule as pr", "prm.pruningRuleId", "pr.id")
-            .select(() => [sql<number>`sum(pr."tokensInText")::int`.as("totalMatchTokens")])
-            .as("totalMatchTokens"),
         ])
         .executeTakeFirst();
 
-      if (!trainingEntryStats) return error("No training data found");
+      if (!datasetEntryStats) return;
+
+      let totalMatchTokens = 0;
+
+      if (pruningRuleIds.length > 0) {
+        const pruningRuleStats = await constructDatasetEntryFiltersQuery({ filters, datasetId })
+          .where("de.split", "=", "TRAIN")
+          .where("de.output", "is not", null)
+          .innerJoin("PruningRuleMatch", (join) =>
+            join
+              .onRef("PruningRuleMatch.datasetEntryId", "=", "de.id")
+              .on("PruningRuleMatch.pruningRuleId", "in", pruningRuleIds),
+          )
+          .leftJoin("PruningRule as pr", "PruningRuleMatch.pruningRuleId", "pr.id")
+          .select(sql<number>`sum(pr."tokensInText")::int`.as("totalMatchTokens"))
+          .executeTakeFirst();
+
+        totalMatchTokens = pruningRuleStats?.totalMatchTokens || 0;
```

Comment:
Since the only thing we care about here is totalMatchTokens, I would set that directly using a `.then` statement.

i.e.
```
let totalMatchTokens = await
...
.executeTakeFirst()
.then((stats) => stats.totalMatchTokens);
```